server/src/test/java/org/elasticsearch/transport/RemoteConnectionManagerTests.java

```diff
@@ -130,6 +131,11 @@ public Version getVersion() {
             return node.getVersion();
         }
 
+        @Override
+        public TransportVersion getTransportVersion() {
+            return TransportVersion.CURRENT;
```

Comment:
This is not correct, but I don't understand `testGetConnection` getting multiple versions from duplicated calls to `getConnection`

Reply:
It's trying to get a connection to a node to which we are not connected, which falls through to creating a `ProxyConnection` via one of the connected nodes chosen in a round-robin fashion:

https://github.com/elastic/elasticsearch/blob/c89f4d195ff573797972d2d66632db25053388cc/server/src/main/java/org/elasticsearch/transport/RemoteConnectionManager.java#L102

Reply:
So its just using the node version as a marker for which nodes are accessed - so not important here

Reply:
We just need to be sure that we access both nodes, but yes the specific versions are not important.